test/AutoDiff/compiler_crashers_fixed/sr15849-invalid-debug-info.swift

```diff
@@ -0,0 +1,141 @@
+// SR-15849: Mutating functions with control flow can cause assertion failure
+// for conflicting debug variable type.
+
+// RUN: %target-swift-frontend -emit-ir -O -g %s | %FileCheck %s
+// CHECK-LABEL: define internal swiftcc float @"$s4main8TestTypeV24doDifferentiableTimeStep04timeG0ySf_tFTJpSSpSrTA"
+// CHECK: [[SELF:%.*]] = alloca %T4main8TestTypeV06ManualB7TangentV, align 4
+// CHECK: call void @llvm.dbg.declare(metadata %T4main8TestTypeV06ManualB7TangentV* [[SELF]]
```

Comment:
should this check for the removed OP_deref?

Reply:
Here is the problem: the op_deref is on SIL level, however, the assertion is only triggered in verifier during LLVM IR codegen and optimized further on. So, essentially w/o the patch we're just having a crash.